ractor/src/port/output/tests.rs

```diff
@@ -157,3 +157,182 @@ async fn test_50_receivers() {
+            Ok(state)
+        }
+
+        async fn handle(
+            &self,
+            _myself: ActorRef<Self::Msg>,
+            message: Self::Msg,
+            state: &mut Self::State,
+        ) -> Result<(), ActorProcessingErr> {
+            match message {
+                PlusSubscriberMessage::Plus(value) => {
+                    let mut state = state.lock().await;
+                    *state += value;
+                }
+            }
+            Ok(())
+        }
+    }
+
+    #[derive(Debug)]
+    enum MulSubscriberMessage {
+        Mul(u8),
+    }
+    impl From<u8> for MulSubscriberMessage {
+        fn from(value: u8) -> Self {
+            MulSubscriberMessage::Mul(value)
+        }
+    }
+
+    struct MulSubscriber;
+    #[cfg_attr(feature = "async-trait", crate::async_trait)]
+    impl Actor for MulSubscriber {
+        type State = Arc<Mutex<u8>>;
+        type Msg = MulSubscriberMessage;
+        type Arguments = Self::State;
+
+        async fn pre_start(
+            &self,
+            _myself: ActorRef<Self::Msg>,
+            state: Self::State,
+        ) -> Result<Self::State, ActorProcessingErr> {
+            Ok(state)
+        }
+
+        async fn handle(
+            &self,
+            _myself: ActorRef<Self::Msg>,
+            message: Self::Msg,
+            state: &mut Self::State,
+        ) -> Result<(), ActorProcessingErr> {
+            match message {
+                MulSubscriberMessage::Mul(value) => {
+                    let mut state = state.lock().await;
+                    *state *= value;
+                }
+            }
+            Ok(())
+        }
+    }
+
+    #[crate::concurrency::test]
+    #[tracing_test::traced_test]
+    async fn test_output_port_subscriber() {
+        let (number_publisher_ref, number_publisher_handler) =
+            Actor::spawn(None, NumberPublisher, ()).await.unwrap();
+
+        let plus_result = Arc::new(Mutex::new(0));
+        let (plus_subcriber_ref, plus_subscriber_handler) =
+            Actor::spawn(None, PlusSubscriber, plus_result.clone())
+                .await
+                .unwrap();
+
+        let mul_result = Arc::new(Mutex::new(1));
+        let (mul_subcriber_ref, mul_subscriber_handler) =
+            Actor::spawn(None, MulSubscriber, mul_result.clone())
+                .await
+                .unwrap();
+
+        cast!(
+            number_publisher_ref,
+            NumberPublisherMessage::Subscribe(Box::new(plus_subcriber_ref.clone()))
+        )
+        .unwrap();
+        cast!(
+            number_publisher_ref,
+            NumberPublisherMessage::Subscribe(Box::new(mul_subcriber_ref.clone()))
+        )
+        .unwrap();
+
+        cast!(number_publisher_ref, NumberPublisherMessage::Publish(2)).unwrap();
+        cast!(number_publisher_ref, NumberPublisherMessage::Publish(3)).unwrap();
+
+        crate::concurrency::sleep(Duration::from_millis(50)).await;
+
+        assert_eq!(2 + 3, *plus_result.lock().await);
```

Comment:
small nit: Normally passing an `Arc<Mutex<_>>` breaks the whole idea of an actor. We'd typically have the actor own the state itself, and use an `RpcReplyPort` to query the actor for the latest state value snapshot, which removes all concurrency concerns like race conditions, serialized updates, etc.

Reply:
I was aware of that but this was just for testing purposes but you are right, specially to avoid teaching though test that you can avoid Mutex, I'll move it to rpc.